src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { Dispatch, SetStateAction } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem';
+import { FilterCompleted } from '../../types/FilterCompleted';
+
+type Props = {
+  todos: Todo[];
+  setTodos: Dispatch<SetStateAction<Todo[]>>;
+  setError: (error: boolean) => void;
+};
+
+export const TodoList: React.FC<Props> = React.memo(({
+  todos,
+  setTodos,
+  setError,
+}) => {
+  const { pathname } = useLocation();
+
+  const visibleTodos = todos.filter((todo) => {
+    switch (pathname) {
+      case FilterCompleted.ACTIVE:
+        return !todo.completed;
+      case FilterCompleted.COMPLETED:
+        return todo.completed;
+      default:
+        return todos;
+    }
+  });
```

Comment:
//[HINT] if you need to filter your items by selected filter - use switch/case statement in correct way.
// Perform the necessary filtering only on a case-by-case basis.

// GOOD

const filteredCards = useMemo(() => {
  switch (filter) {
    case CardFilters.All: {
      return cards;
    }

    case CardFilters.Active: {
      return cards.filter(({ deletedDate }) => !deletedDate);
    }

    case CardFilters.Deleted: {
      return cards.filter(({ deletedDate }) => Boolean(deletedDate));
    }

    default:
      return [];
  }
}, [cards, filter]);

// Don't filter your array right away with switch/case inside your callback.
// Because in case CardFilters.All your program just iterate an array and doesn't do any useful actions. So it's a wasting of memory and CPU (very bad)

// BAD

const filteredCards = useMemo(() => {
  return cards.filter(({ deletedDate }) => {
    switch (filter) {
      case CardFilters.All: {
        return true;
      }

      case CardFilters.Active: {
        return !deletedDate;
      }

      case CardFilters.Deleted: {
        return Boolean(deletedDate);
      }

      default:
        return false;
    }
  });
}, [cards, filter]);